pennylane_lightning_gpu/src/tests/Test_StateVectorCudaManaged_NonParam.cpp

```diff
@@ -830,3 +830,103 @@ TEMPLATE_TEST_CASE("StateVectorCudaManaged::applyCSWAP",
         }
     }
 }
+
+TEMPLATE_TEST_CASE("StateVectorCudaManaged::Hamiltonian_expval",
+                   "[StateVectorCudaManaged_Nonparam]", double) {
+    using cp_t = std::complex<TestType>;
+    const std::size_t num_qubits = 3;
+    SECTION("GetExpectionIdentity") {
+        SVDataGPU<TestType> svdat{num_qubits};
+        std::vector<size_t> wires{0, 1, 2};
+
+        CHECK(svdat.sv.getDataVector()[0] == cp_t{1, 0});
+        svdat.cuda_sv.applyHadamard({0}, false);
+        svdat.cuda_sv.applyCNOT({0, 1}, false);
+        svdat.cuda_sv.applyCNOT({1, 2}, false);
+
+        size_t matrix_dim = static_cast<size_t>(1U) << num_qubits;
+        std::vector<cp_t> matrix(matrix_dim * matrix_dim);
+
+        for (size_t i = 0; i < matrix.size(); i++) {
+            if (i % matrix_dim == i / matrix_dim)
+                matrix[i] = std::complex<TestType>(1, 0);
+            else
+                matrix[i] = std::complex<TestType>(0, 0);
+        }
+
+        auto results = svdat.cuda_sv.expval(wires, matrix);
+
+        cp_t expected(1, 0);
+        CHECK(expected.real() == Approx(results.x).epsilon(0.001));
+        CHECK(expected.imag() == Approx(results.y).epsilon(0.001));
+    }
+
+    SECTION("GetExpectionRandomMatrix") {
+        using cp_t = std::complex<TestType>;
+        const std::size_t num_qubits = 3;
+        std::vector<cp_t> init_state{{0.0, 0.0}, {0.0, 0.1}, {0.1, 0.1},
+                                     {0.1, 0.2}, {0.2, 0.2}, {0.3, 0.3},
+                                     {0.3, 0.4}, {0.4, 0.5}};
+        SVDataGPU<TestType> svdat{num_qubits};
+        svdat.cuda_sv.CopyHostDataToGpu(init_state.data(), init_state.size());
+        std::vector<size_t> wires{0, 1, 2};
+        std::vector<cp_t> matrix{
+            {0.5, 0.0},  {0.2, 0.5},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.5, 0.0},  {0.2, 0.5},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.5, 0.0},  {0.2, 0.5},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.5, 0.0},  {0.2, 0.5},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0}};
+
+        auto results = svdat.cuda_sv.expval(wires, matrix);
+
+        cp_t expected(1.263000, -1.011000);
+
+        CHECK(expected.real() == Approx(results.x).epsilon(0.001));
+        CHECK(expected.imag() == Approx(results.y).epsilon(0.001));
```

Comment:
That epsilon is a little large. Can we reduce it to `1e-7` and still have a passing test?